### PR TITLE
ci: add Node.js v13 nightly to test matrix

### PR DESCRIPTION
**Pull request by watson**
~~Depends on #1277~~

- Adds Node.js 13 in preparation for the upcoming release.
- Removes Node.js 6 and 11, as no new builds will ever be created for those versions.

I ran a test cron job on Travis with these changes to see if everything worked, and it's all green: https://travis-ci.org/elastic/apm-agent-nodejs/builds/570849170

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -173,7 +173,29 @@ pipeline {
...
+          steps {
+            withGithubNotify(context: 'Nightly No Async Hooks Test', tab: 'tests') {
+              deleteDir()
+              unstash 'source'
+              dir("${BASE_DIR}"){
+                script {
+                  def node = readYaml(file: '.ci/.jenkins_nightly_nodejs.yml')
                   def parallelTasks = [:]
                   node['NODEJS_VERSION'].each{ version ->
                     parallelTasks["Node.js-${version}-nightly"] = generateStep(version: version, edge: true)
```
Comment (v1v) [inline Jenkinsfile]:
```suggestion
                    parallelTasks["Node.js-${version}-nightly-no_async_hooks"] = generateStep(version: version, edge: true)
```

File: Jenkinsfile [hunk 2]
```diff
@@ -173,7 +173,29 @@ pipeline {
...
+                  parallel(parallelTasks)
+                }
+              }
+            }
+          }
+        }
+        stage('Nightly Test - No async hooks') {
+          agent { label 'docker && immutable' }
+          environment {
+            NVM_NODEJS_ORG_MIRROR = "https://nodejs.org/download/nightly/"
```
Comment (v1v) [inline Jenkinsfile]:
```suggestion
            NVM_NODEJS_ORG_MIRROR = "https://nodejs.org/download/nightly/"
            ELASTIC_APM_ASYNC_HOOKS = "false"
```

File: Jenkinsfile [hunk 3]
```diff
@@ -213,12 +235,12 @@ pipeline {
...
+            withGithubNotify(context: 'RC No Async Hooks Test', tab: 'tests') {
               deleteDir()
               unstash 'source'
               dir("${BASE_DIR}"){
                 script {
-                  def node = readYaml(file: '.ci/.jenkins_edge_nodejs.yml')
+                  def node = readYaml(file: '.ci/.jenkins_rc_nodejs.yml')
                   def parallelTasks = [:]
                   node['NODEJS_VERSION'].findAll{ it != '6' }.each{ version ->
                     parallelTasks["Node.js-${version}-nightly-no_async_hooks"] = generateStep(version: version, edge: true)
```
Comment (v1v) [inline Jenkinsfile]:
```suggestion
                    parallelTasks["Node.js-${version}-rc-no_async_hooks"] = generateStep(version: version, edge: true)
```

**Comment by watson** [comment]
@v1v Thanks for the review. This PR depends on #1277 so it contains the same commits as #1277, so we'll fix the issues you raised there and then when that PR lands, we'll rebase this PR against master 😃

**Pull request merged**